README.md

```diff
@@ -61,6 +61,48 @@ Implementation details
   * Optional runtime blinding which attempts to frustrate differential power analysis.
   * The precomputed tables add and eventually subtract points for which no known scalar (secret key) is known, preventing even an attacker with control over the secret key used to control the data internally.
 
+Obtaining and verifying
+-----------------------
+
+The git tag for each release (e.g. `v0.6.0`) is GPG-signed by one of the maintainers.
+For a fully verified build of this project, it is recommended to obtain this repository
+via git, obtain the GPG keys of the signing maintainer(s), and then verify the release
+tag's signature using git.
+
+This can be done with the following steps:
+
+1. Check the latest release on the [Releases
+   page](https://github.com/bitcoin-core/secp256k1/releases). Determine the signing GPG ID
+   by clicking the green icon next to the tag name. For example, in the case of v0.6.0,
+   this would be `4861DBF262123605`.
+1. Cross-reference this key ID with another source controlled by its owner (e.g.
+   https://x.com/n1ckler).
```

Comment:
Not sure if this step is security theater. Either this repo is legitimate and the GPG keys are correct or it's not and will have arbitrary instructions here.

One possible justification for including this step is educational: it reminds users how to increase assurance that a GPG key they encounter for the first time is correct. In that case I'd suggest to run the fingerprint through a search engine. Using google, this gives somewhat reasonable results for sipa and myself, but not for @real-or-random.

Reply:
It may be overly paranoid, but I don't think it's security theater. The problem we're trying to mitigate here is a singular trust of what's presented on Github. I think definitionally, we need to consult a source that is not Github (or something obtained from Github) in order to get additional assurance that the keys we're verifying against are indeed your keys.

The thought experiment is "if Github is compromised, and the attacker can modify any of its content, will the instructions result in catching that?" I think consulting a second source for GPG IDs is necessary to fulfill that.

Reply:
Ok, so iiuc the practical failure mode this can prevent is when instructions and SECURITY.md are obtained from two different sources somehow (or same source, at different times) and only SECURITY.md is compromised. Seems reasonable.